Orange/widgets/utils/scaling.py

```diff
@@ -92,8 +99,14 @@ def set_data(self, data, skip_if_same=False, no_data=False):
         if data is None:
             return
 
-        self.domain = data.domain
-        self.data = data
+        domain = data.domain
+        self.primitives = data.domain.attributes + data.domain.class_vars + \
+                          tuple([v for v in data.domain.metas if v.is_primitive()])
```

Comment:
You could use `chain(data.domain, (v for v in data.domain.metas if v.is_primitive()))`.

If you keep it as it is: don't use `tuple([...])`: just use generators, `tuple(...)`, without constructing a list first.